### ISSUE TITLE
🔧 More unit tests, integration test, and doc tests

### ISSUE THREAD

**Issue by neon-mmd** [neon-mmd/websurfx#179]
Add more unit tests, integration tests and doc tests to allow better maintainability and testing of the code.

**Comment by neon-mmd** [comment]
@xffxff are you interested in working on this issue? If you are, then I will assign this issue to you 🙂 .

**Comment by xffxff** [comment]
> @xffxff are you interested in working on this issue? If you are, then I will assign this issue to you 🙂 .

I'm happy to help with this, but there's no need to assign this issue specifically to me. I believe that improving tests is a never-ending task that requires the contributions of all of us. Everyone is welcome to work on this, and this issue could also be a good first issue for those who wish to begin contributing

**Comment by willser** [comment]
I would like to add some ut for understanding this repo.Do you have any suggestion such as where I should start.

**Comment by neon-mmd** [comment]
> I would like to add some ut for understanding this repo.Do you have any suggestion such as where I should start.

Yes sure :slightly_smiling_face:, I think first I would suggest you to go through the [developer](https://github.com/neon-mmd/websurfx/blob/rolling/docs/developing.md) docs to better understand the structure of the project. Then I would recommend you to start writing tests first for covering all the routes in project like for search route, etc. to ensure that all the routes are returning data as expected. Then you can explore other functions which could benefit from writing tests for it. :slightly_smiling_face: 

Also, since you want to work on this issue, so I am assigning this issue to you. You may start working on it right away :rocket: :slightly_smiling_face: .

**Comment by neon-mmd** [comment]
Willser, it has been over a month, any progress on this issue so far? We would like to know :slightly_smiling_face: .

**Comment by neon-mmd** [comment]
@willser, it has been over a month, any progress on this issue so far? We would like to know :slightly_smiling_face: .

**Comment by willser** [comment]
> any progress on this issue so far?

@neon-mmd I am so sorry,I am very busy for my work recently,I am not sure what time I can start this issue.Sorry. 😢

**Comment by neon-mmd** [comment]
> > any progress on this issue so far?
> 
> @neon-mmd I am so sorry,I am very busy for my work recently,I am not sure what time I can start this issue.Sorry. 😢

Sorry for the late reply.
Ok sure, no problem take your time but we would suggest keeping us updated on this issue. Also, if you are unable to work on this issue at all, then we would suggest instead maybe un-assigning yourself from the issue and you can then explore some other quick to work on issue in the project. :slightly_smiling_face:
